docs/source/fetching/persisted-queries.mdx

```diff
@@ -0,0 +1,176 @@
+This allows the PQL to act as a safelist of trusted operations made by your first-party apps. As such, persisted queries is a security feature as much as a performance one.
+
+With APQs, if the server can't find the operation ID the client provides, the server returns an error indicating that it needs the full operation string. If the Apollo iOS client receives this error, it automatically retries the operation with the full operation string.
+
+If you _only_ want to improve request latency and bandwidth usage, APQ addresses your use case. If you _also_ want to secure your supergraph with operation safelisting, you should preregister trusted operations in a PQL.
+
+For more details on differences between persisted queries and APQ, see the [GraphOS persisted queries documentation](/graphos/operations/persisted-queries#differences-from-automatic-persisted).
+
+### Requirements
+
+Persisted queries is currently in [preview](/resources/product-launch-stages#preview) and has the following requirements:
+- Apollo iOS (v1.3.1+)
+- [Apollo Router](/router) (v1.25.0+)
+- [GraphOS Enterprise plan](/graphos/enterprise/)
+
+You can use APQ with the following versions of Apollo iOS, Apollo Server, and Apollo Router:
+- Apollo iOS (v1.0.0+)
+- [Apollo Server](/apollo-server/) (v1.0.0+)
+- [Apollo Router](/router) (v0.1.0+)
+
+> **Note:** You can use _either_ Apollo Server _or_ Apollo Router for APQs. They don't need to be used together.
+
+## Implementation steps
+
+Both persisted queries and APQs require you to configure code generation and how your client makes requests. If you intend to use persisted queries for safelisting, you also need to generate an operations manifest.
+
+We recommend you follow this order while implementing: 
+
+1. **For persisted queries only**: Generate and publish operation manifests
+2. **For both persisted queries and APQs**: Configure generated operation models
+3. **For both persisted queries and APQs**: Enable persisted queries on `ApolloClient`
+
+The rest of this article details these steps.
+
+Persisted queries also requires you to create and link a PQL, and to configure your router. For more information on the other configuration aspects of persisted queries, see the [GraphOS persisted queries documentation](/graphos/operations/persisted-queries).
+
+### 1. Generate operation manifests
+
+> This step is only required for implementing safelisting with persisted queries, it is _not_ for required APQs.
+
+An operation manifest acts as a safelist of trusted operations the [Apollo Router](/router/) can check incoming requests against. You can generate operation manifests in two ways:
+- by running the [`generate`](./../code-generation/codegen-cli#generate) or [`generate-operation-manifest`](./../code-generation/codegen-cli#generate-operation-manifest) CLI command with `--path` and `--version` arguments
+    - `path` is where to write the manifest file
```

Comment:
I assumed that the `path` and `version` arguments would be used in the same was as they are in the config.json, but looking at the [`generate-operation-manifest`](https://www.apollographql.com/docs/ios/code-generation/codegen-cli#generate-operation-manifest) docs, it looks like `path` is where to read the config from (so you have to update the config anyway?) and the `version` option just spits out the CLI version?

Reply:
We had some discussions around the CLI command and config json yesterday and there are going to be some changes to it that I am working on today so this documentation will be updated some based on that and hopefully everything will be more clear.

Reply:
Thank you!